theano/compile/builders.py

```diff
@@ -133,6 +135,59 @@ def perform(self, node, inputs, outputs):
             # we wont need this copy anymore
             output[0] = variable.copy()
 
+    def connection_pattern(self, node):
```

Comment:
This function appears to do the exact same thing (infer the connection pattern from a set of 'input variables' and a set of 'output variables') as the one in the class `Scan` (in scan_op.py). If this is indeed the case, we should avoid the code duplication and have both Scan and OpFromGraph rely on the same implementation which would be in a central location (perhaps somewhere in theano.gof since this is where the class FunctionGraph is defined).